lib/ddtrace/contrib/grpc/datadog_client_interceptor.rb

```diff
@@ -0,0 +1,53 @@
+require 'grpc'
+require 'ddtrace/propagation/grpc_propagator'
+
+module Datadog
+  module Contrib
+    module GRPC
+      # The DatadogClientIntercptor implements the basic interface
+      # for gRPC middleware. When you declare Datadog usage of the gRPC
+      # module, the interceptor must be explicitly declared for usage.
+      class DatadogClientInterceptor < ::GRPC::ClientInterceptor
+        def request_response(request: nil, call: nil, method: nil, metadata: nil)
+          trace_client('client', metadata) { yield }
+        end
+
+        def client_streamer(requests: nil, call: nil, method: nil, metadata: nil)
+          trace_client('client', metadata) { yield }
+        end
+
+        def server_streamer(request: nil, call: nil, method: nil, metadata: nil)
+          trace_client('client', metadata) { yield }
+        end
+
+        def bidi_streamer(requests: nil, call: nil, method: nil, metadata: nil)
+          trace_client('client', metadata) { yield }
+        end
+
+        private
+
+        def ddtracer
+          @ddtracer ||= @options.fetch(:ddtracer, Datadog.tracer)
```

Comment:
There is a `tracer` option on the gRPC configuration. Should we replace `Datadog.tracer` with `Datadog.configuration[:grpc][:tracer]`?